android/src/main/java/com/facebook/flipper/plugins/databases/DatabasesManager.java

```diff
@@ -0,0 +1,202 @@
+package com.facebook.flipper.plugins.databases;
+
+import android.util.SparseArray;
+import com.facebook.flipper.core.FlipperArray;
+import com.facebook.flipper.core.FlipperConnection;
+import com.facebook.flipper.core.FlipperObject;
+import com.facebook.flipper.core.FlipperReceiver;
+import com.facebook.flipper.core.FlipperResponder;
+import com.facebook.flipper.plugins.databases.DatabaseDriver.DatabaseExecuteSqlResponse;
+import com.facebook.flipper.plugins.databases.DatabaseDriver.DatabaseGetTableDataResponse;
+import com.facebook.flipper.plugins.databases.DatabaseDriver.DatabaseGetTableStructureResponse;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class DatabasesManager {
+
+    private final static String DATABASE_LIST_COMMAND = "databaseList";
+    private final static String GET_TABLE_DATA_COMMAND = "getTableData";
+    private final static String GET_TABLE_STRUCTURE_COMMAND = "getTableStructure";
+    private final static String EXECUTE_COMMAND = "execute";
+
+    private final List<DatabaseDriver> databaseDriverList;
+
+    private final SparseArray<DatabaseDescriptorHolder> databaseDescriptorHolderSparseArray;
+    private final Set<DatabaseDescriptorHolder> databaseDescriptorHolderSet;
+
+    private FlipperConnection connection;
+
+    public DatabasesManager(List<DatabaseDriver> databaseDriverList) {
+        this.databaseDriverList = databaseDriverList;
+        this.databaseDescriptorHolderSparseArray = new SparseArray<>();
+        this.databaseDescriptorHolderSet = new TreeSet<>(new Comparator<DatabaseDescriptorHolder>() {
+            @Override
+            public int compare(DatabaseDescriptorHolder o1, DatabaseDescriptorHolder o2) {
+                return o1.databaseDescriptor.name().compareTo(o2.databaseDescriptor.name());
+            }
+        });
+    }
+
+    public void setConnection(FlipperConnection connection) {
+        this.connection = connection;
+        if (connection != null) {
```

Comment:
Any reason for allowing empty connections here?

Reply:
Yes, resetting the connection when onDisconnect() is called